hotel/models.py

```diff
@@ -113,10 +120,18 @@ def legal_last_name(self):
         Hotel exports need split legal names, but we don't collect split legal names, so we're going to have to guess.
 
         Returns one of the following:
+            The second part of the legal name, if the legal name starts with the legal first name
             The second part of the legal name after a space, if the legal name has multiple parts
             The last name, if there is no legal name or if the legal name is just one word
         """
-        return self.legal_name.split(' ', 1)[1] if self.legal_name and ' ' in self.legal_name else self.last_name
+        if self.legal_name:
+            legal_name = self.legal_name.strip()
+            legal_first_name = self.legal_first_name.strip()
+            if legal_name.lower().startswith(legal_first_name.lower()):
```

Comment:
Won't this almost always be true? You're checking the first part of the legal name with a value that is often just the first part of the legal name. What is this supposed to be doing?